src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -139,7 +127,7 @@ export class Kernel implements IKernel {
     private _disposed?: boolean;
     private _disposing?: boolean;
     private _ignoreNotebookDisposedErrors?: boolean;
-    private readonly _kernelSocket = new Subject<KernelSocketInformation | undefined>();
+    private readonly _kernelSocket = new ReplaySubject<KernelSocketInformation | undefined>();
```

Comment:
Why is this a ReplaySubject instead of Subject?

Reply:
Because the kernel finishes creation before the kernelsocket is observed. When the controller signs up for it later, it needs the current value. ReplaySocket does that when something first subscribes.